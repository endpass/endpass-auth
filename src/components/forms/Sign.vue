<template>
  <form
    data-test="sign-form"
    @submit.prevent="emitSubmit"
  >
    <form-field v-if="requesterUrl">
      <a
        :href="requesterUrl"
        data-test="requester-url"
      >{{ requesterUrl }}</a>
      {{ $t('components.sign.requestSign') }}
    </form-field>
    <form-field :label="$t('components.sign.requiresSignBy')">
      <message
        :ellipsis="true"
        data-test="account-address"
      >
        {{ account }}
      </message>
    </form-field>
    <!-- <form-field v-if="error">
      <message
        :error="true"
        data-test="error-message"
      >
        {{ error }}
      </message>
    </form-field> -->
    <form-field :label="$t('components.sign.yourPass')">
      <v-input
        v-model="password"
        :placeholder="$t('components.sign.enterPass')"
        :error="error"
        type="password"
      />
    </form-field>
    <template v-if="isTransaction">
      <form-field label="Value">
        <v-input
          v-model="value"
          :disabled="true"
          type="number"
        />
      </form-field>
      <form-field
        v-if="data"
        :label="$t('components.sign.transactionData')"
      >
        <v-input
          v-model="data"
          :disabled="true"
        />
      </form-field>
      <form-field :label="$t('components.sign.transactionGasPrice')">
        <v-input
          v-model="gasPrice"
          type="number"
        />
      </form-field>
      <form-field v-if="labeledGasPricesList">
        <v-content-switcher
          v-model="gasPrice"
          :items="labeledGasPricesList"
          step="0.0000000001"
          min="0"
        />
      </form-field>
      <form-field :label="$t('components.sign.transactionGasLimit')">
        <v-input
          v-model="gasLimit"
          type="number"
          min="0"
        />
      </form-field>
    </template>
    <form-field
      v-else-if="message"
      :label="$t('components.sign.message')"
    >
      {{ message }}
    </form-field>
    <form-controls>
      <v-button
        :disabled="loading || !password"
        :submit="true"
        type="primary"
        data-test="submit-button"
      >
        {{ primaryButtonLabel }}
      </v-button>
      <v-button
        :disabled="!closable || loading"
        skin="quaternary"
        data-test="cancel-button"
        @click="emitCancel"
      >
        {{ $t('global.close') }}
      </v-button>
    </form-controls>
  </form>
</template>

<script>
import Web3 from 'web3';
import get from 'lodash/get';
import { mapGetters } from 'vuex';
import VInput from '@endpass/ui/kit/VInput';
import VButton from '@endpass/ui/kit/VButton';
import VContentSwitcher from '@endpass/ui/kit/VContentSwitcher';
import Message from '@/components/common/Message.vue';
import FormField from '@/components/common/FormField.vue';
import FormControls from '@/components/common/FormControls.vue';

const { hexToUtf8, fromWei } = Web3.utils;

export default {
  name: 'SignForm',

  props: {
    loading: {
      type: Boolean,
      default: false,
    },

    error: {
      type: String,
      default: null,
    },

    request: {
      type: Object,
      default: null,
    },

    closable: {
      type: Boolean,
      default: true,
    },

    isTransaction: {
      type: Boolean,
      default: false,
    },
  },

  data: () => ({
    password: '',
    value: 0,
    data: '',
    gasPrice: 0,
    gasLimit: 21000,
  }),

  computed: {
    ...mapGetters(['labeledGasPricesList']),

    account() {
      return get(this.request, 'address');
    },

    requesterUrl() {
      return get(this.request, 'url');
    },

    message() {
      const hexMessage = get(this.request, 'request.params[1]');

      if (!hexMessage) return null;

      return hexToUtf8(hexMessage);
    },

    transaction() {
      return get(this.request, 'request.params[0]');
    },

    primaryButtonLabel() {
      return !this.loading
        ? this.$i18n.t('global.sign')
        : this.$i18n.t('global.loading');
    },
  },

  methods: {
    emitSubmit() {
      if (!this.password) return;

      if (!this.isTransaction) {
        this.$emit('submit', {
          account: this.account,
          password: this.password,
        });
        return;
      }

      const updatedTransaction = {
        ...this.transaction,
        value: this.value,
        gasPrice: this.gasPrice,
        gasLimit: this.gasLimit,
      };

      if (this.data) {
        Object.assign(updatedTransaction, {
          data: this.data,
        });
      }

      this.$emit('submit', {
        account: this.account,
        password: this.password,
        transaction: updatedTransaction,
      });
    },

    emitCancel() {
      this.$emit('cancel');
    },
  },

  mounted() {
    if (!this.isTransaction) return;

    this.value = fromWei(this.transaction.value);
    this.gasPrice = fromWei(this.transaction.gasPrice);
    this.data = this.transaction.data;
  },

  components: {
    VButton,
    VInput,
    VContentSwitcher,
    Message,
    FormField,
    FormControls,
  },
};
</script>
