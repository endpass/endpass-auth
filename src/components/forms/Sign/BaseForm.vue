<template>
  <div data-test="sign-form">
    <form-field v-if="error">
      <message
        :error="true"
        data-test="account-address"
      >
        {{ error }}
      </message>
    </form-field>
    <form-field v-if="requesterUrl">
      <a
        :href="requesterUrl"
        data-test="requester-url"
      >{{ requesterUrl }}</a>
      {{ $t('components.sign.requestSign') }}
    </form-field>
    <form-field :label="$t('components.sign.requiresSignBy')">
      <message
        :ellipsis="true"
        data-test="account-address"
      >
        {{ account }}
      </message>
    </form-field>
    <form-field :label="$t('components.sign.yourPass')">
      <v-input
        v-model="password"
        v-validate="'required|min:8'"
        :placeholder="$t('components.sign.enterPass')"
        :error="errors.first('password')"
        :data-vv-as="$t('components.sign.passwordField')"
        name="password"
        type="password"
      />
    </form-field>
    <slot />
    <form-controls>
      <v-button
        :disabled="!closable || loading"
        skin="quaternary"
        type="button"
        data-test="cancel-button"
        @click="emitCancel"
      >
        {{ $t('global.close') }}
      </v-button>
      <v-button
        :disabled="loading || !isFormValid"
        type="button"
        data-test="submit-button"
        @click="emitSubmit"
      >
        {{ primaryButtonLabel }}
      </v-button>
    </form-controls>
  </div>
</template>

<script>
import get from 'lodash/get';
import { mapActions } from 'vuex';
import { setWeb3Network } from '@/service/web3';
import VInput from '@endpass/ui/kit/VInput';
import VButton from '@endpass/ui/kit/VButton';
import Message from '@/components/common/Message.vue';
import FormField from '@/components/common/FormField.vue';
import FormControls from '@/components/common/FormControls.vue';

export default {
  name: 'SignBaseForm',

  inject: ['$validator'],

  props: {
    loading: {
      type: Boolean,
      default: false,
    },

    error: {
      type: String,
      default: null,
    },

    request: {
      type: Object,
      default: null,
    },

    closable: {
      type: Boolean,
      default: true,
    },

    isFormValid: {
      type: Boolean,
      default: false,
    },
  },

  data: () => ({
    password: '',
  }),

  computed: {
    account() {
      return get(this.request, 'address');
    },

    network() {
      return get(this.request, 'net');
    },

    requesterUrl() {
      return get(this.request, 'url');
    },

    primaryButtonLabel() {
      return !this.loading
        ? this.$i18n.t('global.sign')
        : this.$i18n.t('global.loading');
    },
  },

  watch: {
    password() {
      if (this.errors.firstById('incorrectPassword')) {
        this.errors.removeById('incorrectPassword');
      }
    },
  },

  methods: {
    ...mapActions(['validatePassword']),

    async emitSubmit() {
      if (!this.isFormValid) return;

      try {
        await this.validatePassword({
          address: this.account,
          password: this.password,
        });
        this.$emit('submit', {
          account: this.account,
          password: this.password,
        });
      } catch (err) {
        this.errors.add({
          id: 'incorrectPassword',
          field: 'password',
          msg: err.message,
        });
      }
    },

    emitCancel() {
      this.$emit('cancel');
    },
  },

  async mounted() {
    await setWeb3Network(this.network);
  },

  components: {
    VButton,
    VInput,
    Message,
    FormField,
    FormControls,
  },
};
</script>
