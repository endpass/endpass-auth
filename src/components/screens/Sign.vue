<template>
  <screen @close="handleWindowClose">
    <v-modal-card
      :loading="!request"
      :is-closable="isDialog"
      @close="handleSignCancel"
    >
      <sign-form
        :loading="loading"
        :request="request"
        :error="error"
        :closable="isDialog"
        @cancel="handleSignCancel"
        @submit="handleSignSubmit"
      />
    </v-modal-card>
  </screen>
</template>

<script>
import { mapActions, mapGetters, mapState } from 'vuex';
import Screen from '@/components/common/Screen';
import VModalCard from '@endpass/ui/kit/VModalCard';
import SignForm from '@/components/forms/Sign';

export default {
  name: 'Sign',

  data: () => ({
    error: null,
  }),

  computed: {
    ...mapState({
      isInited: state => state.core.isInited,
      loading: state => state.core.loading,
      request: state => state.requests.request,
    }),
    ...mapGetters(['isDialog']),
  },

  methods: {
    ...mapActions(['processRequest', 'cancelRequest', 'dialogClose']),

    async handleSignSubmit(res) {
      try {
        await this.processRequest(res.password);
        this.error = null;
      } catch (err) {
        console.log(err);
        this.error = err.message;
      }
    },

    handleSignCancel() {
      this.cancelRequest();
      this.dialogClose();
    },

    handleWindowClose() {
      this.cancelRequest();
      this.dialogClose();
    },
  },

  components: {
    Screen,
    SignForm,
    VModalCard,
  },
};
</script>
