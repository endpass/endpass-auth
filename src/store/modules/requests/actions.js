import i18n from '@/locales/i18n';
import { signChannel } from '@/class/singleton/channels';
import { Answer } from '@/class';
import signerService from '@/service/signer';
import cryptoDataService from '@/service/cryptoData';

const sendResponse = async ({ commit }, payload) => {
  signChannel.put(Answer.createOk(payload));
  commit('changeLoadingStatus', false);
};

const processRequest = async (
  { state, commit, dispatch, getters },
  password,
) => {
  commit('changeLoadingStatus', true);

  const { address, request, net } = state.request;
  const { demoData } = getters;

  let v3KeyStore;

  if (demoData) {
    // eslint-disable-next-line
    v3KeyStore = demoData.v3KeyStore;
  } else {
    v3KeyStore = await dispatch('getAccount', address);
  }

  try {
    const signResult = await signerService.getSignedRequest({
      v3KeyStore,
      request,
      password,
      net,
    });

    dispatch('sendResponse', {
      id: request.id,
      result: signResult,
      jsonrpc: request.jsonrpc,
    });
  } catch (err) {
    console.log(err.message);

    if (err.message.includes('message authentication code mismatch')) {
      commit('changeLoadingStatus', false);

      throw new Error(i18n.t('store.requests.passIncorrect'));
    } else {
      dispatch('sendResponse', {
        id: request.id,
        result: [],
        error: err,
        jsonrpc: request.jsonrpc,
      });
    }
  } finally {
    commit('changeLoadingStatus', false);
  }
};

const recoverMessage = async ({ dispatch }, payload) => {
  const { address, request, net } = payload;

  try {
    const account = await dispatch('getAccount', address);
    const res = await signerService.recoverMessage({ account, request, net });

    return {
      id: request.id,
      result: res,
      jsonrpc: request.jsonrpc,
    };
  } catch (err) {
    return {
      id: request.id,
      result: null,
      error: err,
      jsonrpc: request.jsonrpc,
    };
  }
};

const cancelRequest = ({ state, dispatch }) => {
  const { request } = state.request;

  dispatch('sendResponse', {
    id: request.id,
    error: 'canceled',
    result: [],
  });
};

// TODO: move to external module
const getGasPrice = async (ctx, network) => {
  const prices = await cryptoDataService.getGasPrice(network);

  return prices;
};

export default {
  processRequest,
  recoverMessage,
  sendResponse,
  cancelRequest,
  getGasPrice,
};
