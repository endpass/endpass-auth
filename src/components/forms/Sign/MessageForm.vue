<template>
  <base-form
    :request="request"
    :error="error"
    :loading="loading"
    :closable="closable"
    :is-form-valid="isFormValid"
    @cancel="emitCancel"
    @submit="emitSubmit"
  >
    <form-field
      v-if="message"
      :label="$t('components.sign.requestMessage')"
      data-test="sign-form-message"
    >
      {{ message }}
    </form-field>
  </base-form>
</template>

<script>
import Web3 from 'web3';
import get from 'lodash/get';
import formMixin from '@/mixins/form';
import FormField from '@/components/common/FormField.vue';
import BaseForm from './BaseForm.vue';

const { hexToUtf8 } = Web3.utils;

export default {
  name: 'SignMessageForm',

  props: {
    loading: {
      type: Boolean,
      default: false,
    },

    error: {
      type: String,
      default: null,
    },

    request: {
      type: Object,
      default: null,
    },

    closable: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    message() {
      const method = get(this.request, 'request.method');
      const hexMessage =
        method === 'personal_sign'
          ? get(this.request, 'request.params[0]')
          : get(this.request, 'request.params[1]');

      if (!hexMessage) return null;

      return hexToUtf8(hexMessage);
    },
  },

  methods: {
    async emitSubmit(payload) {
      this.$emit('submit', payload);
    },

    emitCancel() {
      this.$emit('cancel');
    },
  },

  mixins: [formMixin],

  components: {
    FormField,
    BaseForm,
  },
};
</script>
