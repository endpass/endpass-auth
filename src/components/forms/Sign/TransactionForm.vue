<template>
  <base-form
    :request="request"
    :error="error"
    :loading="loading"
    :closable="closable"
    :is-form-valid="isFormValid"
    :title="$t('components.sign.requiresPaymentBy')"
    @cancel="emitCancel"
    @submit="emitSubmit"
  >
    <div data-test="sign-form-transaction-params">
      <form-field :label="$t('components.sign.transactionTo')">
        <v-address :address="transaction.to" />
      </form-field>
      <div class="sign-transaction-form-values">
        <form-field :label="$t('components.sign.transactionValue')">
          <v-input
            v-model="valueToDisplay"
            v-validate="'required|min:0'"
            :error="errors.first('value')"
            :disabled="true"
            :data-vv-as="$t('components.sign.valueField')"
            name="value"
            type="number"
          />
        </form-field>
        <form-field
          :label="
            $t('components.sign.transactionValueFiat', {
              currency: fiatCurrency,
            })
          "
        >
          <v-input
            v-model="valueInFiat"
            :disabled="true"
            name="value"
            type="number"
          />
        </form-field>
      </div>
      <form-field>
        <button
          class="sign-transaction-form-toggler"
          type="button"
          @click="handleToggleAdvancedSettings"
        >
          {{ advancedSettingsTogglerLabel }}
        </button>
      </form-field>
      <div v-show="isAdvancedOptionsVisible">
        <form-field
          v-if="data"
          :label="$t('components.sign.transactionData')"
        >
          <v-input
            v-model="data"
            :disabled="true"
          />
        </form-field>
        <form-field :label="$t('components.sign.transactionGasPrice')">
          <v-input
            v-model="gasPrice"
            v-validate="'required|between:1,100'"
            :error="errors.first('gasPrice')"
            :data-vv-as="$t('components.sign.gasPriceField')"
            name="gasPrice"
            type="number"
            step="1"
            min="1"
            max="100"
          />
        </form-field>
        <form-field
          v-if="labeledGasPricesList"
          :label="$t('components.sign.transactionPriority')"
        >
          <v-content-switcher
            v-model="gasPrice"
            :items="labeledGasPricesList"
          />
        </form-field>
        <form-field :label="$t('components.sign.transactionGasLimit')">
          <v-input
            v-model="gasLimit"
            v-validate="'required|numeric|integer|between:21000,1000000'"
            :error="errors.first('gasLimit')"
            :data-vv-as="$t('components.sign.gasLimitField')"
            name="gasLimit"
            type="number"
            step="1000"
            min="21000"
            max="1000000"
          />
        </form-field>
      </div>
    </div>
  </base-form>
</template>

<script>
import { BigNumber } from 'bignumber.js';
import Web3 from 'web3';
import get from 'lodash/get';
import { mapActions, mapState } from 'vuex';
import formMixin from '@/mixins/form';
import VInput from '@endpass/ui/kit/VInput';
import VContentSwitcher from '@endpass/ui/kit/VContentSwitcher';
import FormField from '@/components/common/FormField.vue';
import VAddress from '@/components/common/VAddress.vue';
import BaseForm from './BaseForm.vue';

const { fromWei, toWei, hexToNumberString } = Web3.utils;

export default {
  name: 'SignTransactionForm',

  props: {
    loading: {
      type: Boolean,
      default: false,
    },

    error: {
      type: String,
      default: null,
    },

    request: {
      type: Object,
      default: null,
    },

    closable: {
      type: Boolean,
      default: true,
    },
  },

  data: () => ({
    gasPrices: null,
    value: '0',
    data: '',
    gasPrice: '0',
    gasLimit: '0',
    ethPrice: '0',
    isAdvancedOptionsVisible: false,
    isInited: false,
  }),

  computed: {
    ...mapState({
      settings: state => state.accounts.settings,
      balance: state => state.accounts.balance,
    }),

    transaction() {
      return get(this.request, 'request.params[0]', null);
    },

    labeledGasPricesList() {
      if (!this.gasPrices) return null;

      return Object.keys(this.gasPrices).reduce((acc, key) => {
        const priority = key.charAt(0).toUpperCase() + key.slice(1);

        return acc.concat({
          label: `${priority} (${this.gasPrices[key]} gwei)`,
          value: this.gasPrices[key].toString(),
        });
      }, []);
    },

    advancedSettingsTogglerLabel() {
      if (this.isAdvancedOptionsVisible) {
        return this.$t('components.sign.closeAdvancedSettings');
      }

      return this.$t('components.sign.openAdvancedSettings');
    },

    fiatCurrency() {
      return get(this.settings, 'fiatCurrency', 'USD');
    },

    valueToDisplay() {
      return BigNumber(this.value).toFixed(6);
    },

    valueInFiat() {
      return BigNumber(this.value)
        .times(this.ethPrice)
        .toFixed(2);
    },

    maxAmount() {
      const { balance, gasPrice, gasLimit } = this;

      if (!this.isInited || !balance || balance === '0') return '0';

      const balanceBN = BigNumber(balance);
      const gasFeeBN = BigNumber(toWei(gasPrice, 'gwei')).times(gasLimit);
      const maxAmountBN = balanceBN.minus(gasFeeBN);
      const maxAmount = fromWei(maxAmountBN.toFixed());

      return maxAmount > 0 ? maxAmount : 0;
    },
  },

  methods: {
    ...mapActions([
      'getGasPrices',
      'getGasLimitByAddress',
      'getEtherPrice',
      'subscribeOnBalanceUpdates',
    ]),

    handleToggleAdvancedSettings() {
      this.isAdvancedOptionsVisible = !this.isAdvancedOptionsVisible;
    },

    async emitSubmit({ account, password }) {
      this.$emit('submit', {
        transaction: {
          ...this.transaction,
          value: this.value,
          gas: this.gasLimit,
          gasLimit: this.gasLimit,
          gasPrice: this.gasPrice,
          data: this.data,
        },
        account,
        password,
      });
    },

    emitCancel() {
      this.$emit('cancel');
    },
  },

  created() {
    this.subscribeOnBalanceUpdates();
  },

  async mounted() {
    const { net } = this.request;
    const { to, value, gasPrice, gas, gasLimit, data } = this.transaction;
    const trxGasLimit = gas || gasLimit;

    this.ethPrice = await this.getEtherPrice(this.fiatCurrency);
    this.gasPrices = await this.getGasPrices(net);

    if (!trxGasLimit) {
      this.gasLimit = await this.getGasLimitByAddress(to);
    } else {
      this.gasLimit = hexToNumberString(trxGasLimit);
    }

    this.value = value ? fromWei(value, 'ether') : '0';
    this.gasPrice = gasPrice ? fromWei(gasPrice, 'gwei') : '1';
    this.data = data || '0x';
    this.isInited = true;
  },

  mixins: [formMixin],

  components: {
    BaseForm,
    VInput,
    VContentSwitcher,
    VAddress,
    FormField,
  },
};
</script>

<style lang="postcss">
.sign-transaction-form-values {
  display: flex;
  align-items: center;
  margin-bottom: 20px;

  & > * {
    flex: 0 0 calc(50% - 10px);
    margin-bottom: 0;
  }

  & > *:first-child {
    margin-right: 20px;
  }
}

.sign-transaction-form-toggler {
  display: inline-block;
  vertical-align: middle;
  font-size: 16px;
  text-decoration: underline;
  border: none;
  background: none;
  cursor: pointer;
  color: #044bda;
}
</style>
