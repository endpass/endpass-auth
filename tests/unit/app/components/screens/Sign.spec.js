import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Sign from '@/components/screens/Sign.vue';
import setupI18n from '@/locales/i18nSetup';
import { requestWithTransaction } from '@unitFixtures/requests';

const localVue = createLocalVue();

localVue.use(Vuex);
const i18n = setupI18n(localVue);

describe('Sign', () => {
  describe('render', () => {
    let store;
    let storeData;
    let wrapper;
    let accountsModule;
    let requestsModule;
    let gasPriceModule;
    let coreModule;

    beforeEach(() => {
      coreModule = {
        state: {
          isInited: true,
          loading: false,
        },
        actions: {
          sendReadyMessage: jest.fn(),
          dialogClose: jest.fn(),
        },
        getters: {
          isDialog: jest.fn(() => true),
        },
      };
      accountsModule = {
        state: {
          accounts: [],
        },
      };
      requestsModule = {
        state: {
          request: {},
        },
        actions: {
          awaitRequestMessage: jest.fn(),
          processRequest: jest.fn(),
          cancelRequest: jest.fn(),
        },
      };
      gasPriceModule = {
        actions: {
          getGasPrice: jest.fn(),
        },
      };
      storeData = {
        modules: {
          accounts: accountsModule,
          core: coreModule,
          requests: requestsModule,
          gasPrice: gasPriceModule,
        },
      };
      store = new Vuex.Store(storeData);
      wrapper = shallowMount(Sign, {
        localVue,
        store,
        i18n,
      });
    });

    describe('render', () => {
      it('should correctly render Sign component', () => {
        expect(wrapper.name()).toBe('Sign');
        expect(wrapper.html()).toMatchSnapshot();
      });
    });

    describe('behavior', () => {
      it('should confirm current request with password', () => {
        // TODO Have troubles with triggering event from stub, solve it when possivble
        wrapper.vm.handleSignSubmit({
          password: 'foo',
        });

        expect(requestsModule.actions.processRequest).toBeCalled();
      });

      it('should cancel current request', () => {
        // TODO Have troubles with triggering event from stub, solve it when possivble
        wrapper.vm.handleSignCancel();

        expect(requestsModule.actions.cancelRequest).toBeCalled();
      });

      it('should not request gas prices if current request not contains transaction', async () => {
        expect.assertions(1);

        await global.flushPromises();

        expect(gasPriceModule.actions.getGasPrice).not.toBeCalled();
      });

      it('should request gas prices if current request contains transaction', async () => {
        expect.assertions(1);

        requestsModule.state.request = requestWithTransaction;

        wrapper = shallowMount(Sign, {
          localVue,
          store,
          i18n,
        });

        await global.flushPromises();

        expect(gasPriceModule.actions.getGasPrice).toBeCalled();
      });
    });
  });
});
