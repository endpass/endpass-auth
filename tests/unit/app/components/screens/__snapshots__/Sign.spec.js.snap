// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign render render should correctly render Sign component 1`] = `
<screen-stub>
  <v-modal-card-stub isclosable="true">
    <sign-form-stub request="[object Object]" closable="true"></sign-form-stub>
  </v-modal-card-stub>
</screen-stub>
`;
